### PR TITLE
OCPBUGS-41834: Add agent installer checks for hosts interface table

### PR DESCRIPTION
**Pull request by bfournie**
Add checks for the host interface table to validate:
- at least one entry is defined in the interface table when using network configuration
- the MAC addresses in the interface table are using the expected format

### CODE REVIEW

File: pkg/asset/agent/manifests/staticnetworkconfig/generator.go [hunk 1]
```diff
@@ -219,20 +220,26 @@ func (s *staticNetworkConfigGenerator) formatNMConnection(nmConnection string) (
...
 			err = multierror.Append(err, fmt.Errorf("failed to validate network yaml for host %d, %w", i, validateErr))
 		}
 	}
 	return err.ErrorOrNil()
 }
 
-func (s *staticNetworkConfigGenerator) validateMacInterfaceName(hostIdx int, macInterfaceMap models.MacInterfaceMap) error {
+func (s *staticNetworkConfigGenerator) validateMacInterfaceTable(hostIdx int, macInterfaceMap models.MacInterfaceMap) error {
+	if len(macInterfaceMap) == 0 {
+		return fmt.Errorf("at least one interface for host %d must be provided", hostIdx)
```
Comment (rwsu) [inline pkg/asset/agent/manifests/staticnetworkconfig/generator.go:233]: Just wanted to check my understanding. We don't validate the interfaces also exist in the raw NMStateConfig, correct?

**Review by rwsu — COMMENTED** [review]
/lgtm

File: pkg/asset/agent/manifests/staticnetworkconfig/generator.go (hunk 1 above)
Comment (bfournie) [reply pkg/asset/agent/manifests/staticnetworkconfig/generator.go:233]: That's correct, this is just for checking the interfaces in the MAC Interface table.

**Pull request merged**